# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1478533903 28800
#      Mon Nov 07 07:51:43 2016 -0800
# Branch stable
# Node ID 838d52a3283e9c9f28f7d22d206988231b3b60f0
# Parent  125391edc71b4f5cf19cb34774b17266899fbaac
Stop segfault when popen2 called with invalid command (bug #49525).

* octave-popen2.c: Call _exit, rather than exit, when there is an error
in the child process.

diff --git a/liboctave/wrappers/octave-popen2.c b/liboctave/wrappers/octave-popen2.c
--- a/liboctave/wrappers/octave-popen2.c
+++ b/liboctave/wrappers/octave-popen2.c
@@ -201,17 +201,17 @@ octave_popen2 (const char *cmd, char *co
                 perror ("popen2 (child)");
             }
           else
             perror ("popen2 (child)");
         }
       else
         perror ("popen2 (child)");
 
-      exit (0);
+      _exit (127);
     }
   else if (pid > 0)
     {
       // Parent process
 
       close (child_stdin[0]);
       close (child_stdout[1]);
 
